bridge/setu/listener/maticchain.go

```diff
@@ -54,14 +55,42 @@ func (ml *MaticChainListener) Start() error {
 // ProcessHeader - process headerblock from maticchain
 func (ml *MaticChainListener) ProcessHeader(newHeader *types.Header) {
 	ml.Logger.Debug("New block detected", "blockNumber", newHeader.Number)
+
+	// check and send span task
+	go ml.checkAndSendSpanTask(newHeader)
```

Comment:
This will fetch the last span very frequently. We should do span poll as we were doing before or optimize the last span fetch using cache.

Reply:
Added cache